### PR TITLE
fix(frontend): Fix magic shelves being wrong.

### LINKED ISSUE #2868: [Bug] Subtitle field does not work in Magic Shelf rules
### What Went Wrong?

Using Subtitle in a Magic Shelf rule does not work. I've tried a magic shelf with only a subtitle rule and in combination with others rules. I've tried most operators: Equals, Includes Any, Contains, Starts With, etc. 

### How Can We Reproduce This?

1. Go to Magic Shelves in the left side bar and click the + to start creating a new magic shelf
2. Add a rule using Subtitle as the field, Equals as the operator, and a known subtitle for an existing book as the value
3. Enter a name for the shelf and save
4. Navigate to the new shelf

Result: The shelf will be empty

### What Should Have Happened?

Books with the matching subtitle should appear in the magic shelf.

### Screenshots or Error Messages (Optional)

Image

Image

Image

### Any Ideas on How to Fix This? (Optional)

_No response_

### Your Setup

- Booklore version: 2.0.0
- OS: Windows 10, Android 16
- Browser: Chrome, Firefox
- Installation: Docker
- Storage Type: Local drive

### Before Submitting

- [x] I've searched existing issues and confirmed this bug hasn't been reported yet

### PR DESCRIPTION
**Pull request by JVT038**
## 📝 Description

**Linked Issue:** Fixes #2868 

> **Required.** Every PR must reference an approved issue. If no issue exists, [open one](https://github.com/booklore-app/booklore/issues/new) and wait for maintainer approval before submitting a PR. Unsolicited PRs without a linked issue will be closed.

## 🏷️ Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement to existing feature
- [ ] Refactor (no behavior change)
- [ ] Breaking change (existing functionality affected)
- [ ] Documentation update

## 🔧 Changes

- Add withDescription parameter and set it to true, so that the backend doesn't strip tne metadata fields.

## 🧪 Testing (MANDATORY)

> **PRs without this section filled out will be closed.** "Tests pass" or "Tested locally" is not sufficient. You must provide specifics.

**Manual testing steps you performed:**
1. Open booklore
2. See magic shelves properly filled

**Regression testing:**

All frontend tests still pass.

**Edge cases covered:**
N/A

**Test output:**

<details>
<summary>Backend test output (<code>./gradlew test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

<details>
<summary>Frontend test output (<code>ng test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

## 📸 Screen Recording / Screenshots (MANDATORY)

> Every PR must include a **screen recording or screenshots** showing the change working end-to-end in a running local instance (both backend and frontend). This means you must have actually built, run, and tested the code yourself. PRs without visual proof will be closed without review.

---

## ✅ Pre-Submission Checklist

> **All boxes must be checked before requesting review.** Incomplete PRs will be closed without review. No exceptions.

- [x] This PR is linked to an approved issue
- [x] Code follows project style guidelines and conventions
- [x] Branch is up to date with `develop` (merge conflicts resolved)
- [x] I ran the full stack locally (backend + frontend + database) and verified the change works
- [x] Automated tests added or updated to cover changes (backend **and** frontend)
- [x] All tests pass locally and output is pasted above
- [ ] Screen recording or screenshots are attached above proving the change works
- [x] PR is a single focused change (one bug fix OR one feature, not multiple unrelated changes)
- [x] PR is reasonably scoped (PRs over 1000+ changed lines will be closed, split into smaller PRs)
- [x] No unsolicited refactors, cleanups, or "improvements" are bundled in
- [ ] Flyway migration versioning is correct _(if schema was modified)_
- [ ] Documentation PR submitted to [booklore-docs](https://github.com/booklore-app/booklore-docs) _(if user-facing changes)_

### 🤖 AI-Assisted Contributions

> **If any part of this PR was generated or assisted by AI tools (Copilot, Claude, ChatGPT, etc.), all items below are mandatory.** You are fully responsible for every line you submit. "The AI wrote it" is not an excuse, and AI-generated PRs that clearly haven't been reviewed are the #1 reason PRs get closed.

- [ ] I have read and understand every line of this PR and can explain any part of it during review
- [ ] I personally ran the code and verified it works (not just trusted the AI's output)
- [ ] PR is scoped to a single logical change, not a dump of everything the AI suggested
- [ ] Tests validate actual behavior, not just coverage (AI-generated tests often assert nothing meaningful)
- [ ] No dead code, placeholder comments, `TODO`s, or unused scaffolding left behind by AI
- [ ] I did not submit refactors, style changes, or "improvements" the AI suggested beyond the scope of the issue

---

### CODE REVIEW

**Comment by balazs-szucs** [comment]
Hi,

thank you for your submission. Please note that we require all PRs to use our standard message template and include proof of task completion. Could you please update your PR description to meet these guidelines? Thanks!

File: booklore-ui/src/app/features/book/service/book.service.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class BookService {
   }
 
   private fetchBooks(): Observable<Book[]> {
-    return this.http.get<Book[]>(this.url).pipe(
+    return this.http.get<Book[]>(this.url, {params: {withDescription: true}}).pipe(
```
Comment (acx10) [inline booklore-ui/src/app/features/book/service/book.service.ts:75]: hey, the issue is real but this is a pretty heavy fix. `withDescription: true` disables `stripFieldsForListView` entirely, which means the full `description` text (often thousands of chars per book) gets sent for every book on initial load. for large libraries that's a lot of extra payload.

most of the fields that get stripped are lightweight (subtitle, ratings, seriesTotal, etc.) and there's no reason to lump them in with `description`. a better fix would be on the backend: keep stripping `description` (the actually heavy field) but stop stripping the smaller metadata fields so magic shelf rules can use them without bloating the response.
